nomos-services/mix/src/lib.rs

```diff
@@ -77,73 +79,71 @@ where
         loop {
             tokio::select! {
-                Some(msg) = incoming_message_stream.next() => {
-                    tracing::debug!("Received message from mix backend. Sending it to Processor");
-                    if let Err(e) = processor_inbound_sender.send(msg) {
-                        tracing::error!("Failed to send incoming message to processor: {e:?}");
-                    }
-                }
-                Some(msg) = emission_receiver.recv() => {
-                    tracing::debug!("Emitting message to mix network");
+                Some(msg) = persistent_transmission_messages.next() => {
                     backend.publish(msg).await;
                 }
-                Some(msg) = fully_unwrapped_message_receiver.recv() => {
-                    tracing::debug!("Broadcasting fully unwrapped message");
-                    match wire::deserialize::<NetworkMessage<Network::BroadcastSettings>>(&msg) {
-                        Ok(msg) => {
-                            network_adapter.broadcast(msg.message, msg.broadcast_settings).await;
-                        },
-                        _ => tracing::error!("unrecognized message from mix backend")
+                // Already processed blend messages
+                Some(msg) = blend_messages.next() => {
+                    match msg {
+                        MixOutgoingMessage::Outbound(msg) => {
+                            if let Err(e) = persistent_sender.send(msg) {
+                                tracing::error!("Error sending message to persistent stream: {e}");
+                            }
+                        }
+                        MixOutgoingMessage::FullyUnwrapped(msg) => {
+                            tracing::debug!("Broadcasting fully unwrapped message");
+                            match wire::deserialize::<NetworkMessage<Network::BroadcastSettings>>(&msg) {
+                                Ok(msg) => {
+                                    network_adapter.broadcast(msg.message, msg.broadcast_settings).await;
+                                },
+                                _ => {
+                                    tracing::error!("unrecognized message from mix backend");
+                                }
+                            }
+                        }
                     }
                 }
-                Some(msg) = service_state.inbound_relay.recv() => {
-                    Self::handle_service_message(msg, &new_message_sender);
+                Some(msg) = local_messages.next() => {
+                    match cryptographic_processor.wrap_message(&msg) {
+                        Ok(wrapped_message) => {
+                            backend.publish(wrapped_message).await;
+                        }
+                        Err(e) => {
+                            tracing::error!("Failed to wrap message: {:?}", e);
+                        }
+                    }
```

Comment:
Local messages are encoded with the cryptographic processor and published directly.

Reply:
Local messages should be passed to the persistent transmission (tier 1). It bypasses only tier 2. Tier 1 is essential for any message to be published to the mix network.
https://www.notion.so/Nomos-Blend-Network-Tier-2-Message-Blending-Module-1208f96fb65c80e5bcb9df6e27472339?pvs=4#1248f96fb65c8081b58ffcca7305de88

Reply:
> Thank you so much for all of your help!! We have only one thing to change. Except it, everything looks good!
> 
> Local messages should be passed to the persistent transmission (tier 1). It bypasses only tier 2. Tier 1 is essential for any message to be published to the mix network. https://www.notion.so/Nomos-Blend-Network-Tier-2-Message-Blending-Module-1208f96fb65c80e5bcb9df6e27472339?pvs=4#1248f96fb65c8081b58ffcca7305de88
> 
> Maybe, like this?

Yeah, totally skipped this. Good catch!!